### ISSUE TITLE
Babel support

### ISSUE THREAD

**Issue by Globegitter** [nexe/nexe#109]
In most of my projects that are not performance critical I am using [Babel](https://github.com/babel/babel). Just tried out nexe on one of these projects but it has issues at least with ES7 syntax, e.g. module.exports = async function (cmdArgs) {.

The file that I am building has babel defined as follows:

``` js
//run.js
require('babel/register')({
  experimental: true,
  loose: true
});

var nodegitFile = require('./nodegit');

nodegitFile(['tests/fixtures/example.txt']);
```

and running `node run.js` works fine.

The error I am getting on `nexe -i run.js` is:

``` js
----> bundle project/blame-history/run.js
events.js:141
      throw er; // Unhandled 'error' event
            ^
Error: Parsing file project/blame-history/nodegit.js: Unexpected token (5:23)
    at Deps.parseDeps (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/index.js:418:28)
    at fromSource (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/index.js:361:48)
    at /Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/index.js:356:17
    at ConcatStream.<anonymous> (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/node_modules/concat-stream/index.js:32:43)
    at emitNone (events.js:72:20)
    at ConcatStream.emit (events.js:163:7)
    at finishMaybe (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:460:14)
    at endWritable (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:469:3)
    at ConcatStream.Writable.end (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:436:5)
    at Transform.onend (/Users/markus/npm/lib/node_modules/nexe/node_modules/module-deps/node_modules/readable-stream/lib/_stream_readable.js:523:10)
```

Any chance to get nexe to support Babel?

**Comment by RainbowDashDC** [comment]
I'll look into this, never heard of babel up until now.

**Comment by Globegitter** [comment]
Awesome thanks - really appreciated.

**Comment by cloudkite** [comment]
+1

**Comment by calebboyd** [comment]
Just a note: I avoided this by using the babel runtime and polyfill.

**Comment by Apercu** [comment]
Any update on this @jaredallard ? Thanks

**Comment by defunctzombie** [comment]
@calebboyd could you elaborate on your workaround?

**Comment by calebboyd** [comment]
Had even more trouble with different pre-processing down the road, along with custom parameters.

Ended up making a [cli tool](https://github.com/calebboyd/xbin) that uses node's native `_third_party_main` module. It doesn't do any bundling or patching of source files -- but just updates the build configuration (`node.gyp`) and adds _your own bundle_

**Comment by jaredallard** [comment]
Yeah, there is a lot of issue with pre-processing. A lot of it has been
addressed in v2. Just not in a working state for usage right now and not
releasing until it's stable. Then it'll be on master and tested for
absolute stability before releasing. 1.0 is a mess right now and not much I
can do but put these issues on backlog for when it's put into v2. (plus
I've had finals and pre-finals for weeks now)

**Comment by defunctzombie** [comment]
For anyone else that stumbles here wondering how to package up newer js here is an approach that will work.

**tl;dr;** use webpack to create a single js file and package that with nexe

Use [webpack](https://webpack.github.io/) to create a single js file starting from some entry point.

Below is a `webpack.config.js` that I used.

```js
const webpack = require('webpack');

module.exports = {
    entry: ['babel-polyfill', './cli.js'],
    output: {
        path: "/tmp",
        filename: 'build.js',
    },
    target: 'node',
    plugins: [
        new webpack.optimize.DedupePlugin(),
        new webpack.optimize.OccurenceOrderPlugin(),
        new webpack.optimize.UglifyJsPlugin(),
    ],
    module: {
        loaders: [{
            test: /\.jsx?$/,
            exclude: /node_modules/,
            loader: 'babel',
            query: {
                plugins: [
                    'transform-class-properties',
                    'transform-async-to-generator',
                    'transform-object-rest-spread',
                ],
                presets: ['es2015'],
            },
        }, {
            test: /\.json$/,
            loader: 'json',
        }],
    },
};
```

You will need to ensure you have the correct dependencies in your `package.json`. Here is an example from mine.

```json
{
  "name": "bagstore-cli",
  "version": "0.0.0",
  "dependencies": {
    "ansi": "0.3.1",
    "babel-cli": "6.9.0",
    "babel-core": "6.9.0",
    "babel-loader": "6.2.4",
    "babel-plugin-transform-async-to-generator": "6.8.0",
    "babel-plugin-transform-class-properties": "6.9.0",
    "babel-plugin-transform-es2015-modules-commonjs": "6.8.0",
    "babel-plugin-transform-object-rest-spread": "6.8.0",
    "babel-preset-es2015": "6.9.0",
    "commander": "2.9.0"
  },
  "devDependencies": {
    "memory-fs": "0.3.0",
    "webpack": "1.13.1"
  },
}
```

In my case `cli.js` was my entrypoint. Webpack will "follow" your import or require expressions and you will end up with a single file `/tmp/build.js` which is all of the code for your CLI app packaged up in a single js file.

With the above config file, run the webpack CLI command per usual. It will produce `/tmp/build.js` which is your entire app as a single js file suitable for running with node. (i.e. `node /tmp/build.js` to test it).

I have not taken the step to actually package it with nexe (since having a single JS file was good enough for my use case) but it should be relatively straight forward to do so with the final js file.

**Comment by jaredallard** [comment]
Sorry everyone finals are a big thing right now, webpack is in v2 so I'll
make sure babel can be used through it and etc

**Comment by calebboyd** [comment]
Bundling is decoupled from nexe in `nexe@beta`
